Source/WebCore/page/Quirks.cpp

```diff
@@ -1515,11 +1515,36 @@ bool Quirks::shouldExposeShowModalDialog() const
 {
     if (!needsQuirks())
         return false;
+
     if (!m_shouldExposeShowModalDialog) {
         auto domain = RegistrableDomain(m_document->url()).string();
         m_shouldExposeShowModalDialog = domain == "pandora.com"_s || domain == "marcus.com"_s;
     }
     return *m_shouldExposeShowModalDialog;
 }
 
+bool Quirks::shouldDisableLazyImageLoadingQuirk() const
+{
+    if (!needsQuirks())
+        return false;
+
+    if (m_shouldDisableLazyImageLoadingQuirk)
+        return m_shouldDisableLazyImageLoadingQuirk.value();
+
+    m_shouldDisableLazyImageLoadingQuirk = false;
+
+    // Images will have a grey box over them when loaded lazily in large percentage of page loads.
+    // This issue is not observed when lazy image loading is disabled, and has been fixed in future Gatsby versions.
+    // This quirk only needs to be applied to IKEA.com when "<meta name="generator" content="Gatsby 4.24.1" />" is present.
+
+    auto* metaElement = m_document->getElementsByTagName("meta"_s)->namedItem("generator"_s);
+    
+    if ((RegistrableDomain(m_document->url()).string() == "ikea.com"_s)
```

Comment:
This check needs to happen earlier. Why are we even traversing the DOM tree to find meta tags if we're not on IKEA?

Reply:
Good point fixed.